e2e/utils.ts

```diff
@@ -1,31 +1,44 @@
+import * as fs from 'fs';
+
 import {chromium} from '@playwright/test';
 
 import {userPrefix, channelPrefix} from './constants';
 import PlaywrightDevPage from './page';
 
+export const headers = {'X-Requested-With': 'XMLHttpRequest'};
+
 export function getChannelNamesForTest() {
     let idx = 0;
     if (process.env.TEST_PARALLEL_INDEX) {
-        idx = parseInt(String(process.env.TEST_PARALLEL_INDEX), 10) * 2;
+        idx = parseInt(String(process.env.TEST_PARALLEL_INDEX), 10) * 3;
     }
-    return [`${channelPrefix}${idx}`, `${channelPrefix}${idx + 1}`];
+    return [`${channelPrefix}${idx}`, `${channelPrefix}${idx + 1}`, `${channelPrefix}${idx + 2}`];
 }
 
 export function getUserIdxForTest() {
     if (process.env.TEST_PARALLEL_INDEX) {
-        return parseInt(String(process.env.TEST_PARALLEL_INDEX), 10) * 2;
+        return parseInt(String(process.env.TEST_PARALLEL_INDEX), 10) * 3;
     }
     return 0;
 }
 
 export function getUsernamesForTest() {
     const idx = getUserIdxForTest();
-    return [`${userPrefix}${idx}`, `${userPrefix}${idx + 1}`];
+    return [`${userPrefix}${idx}`, `${userPrefix}${idx + 1}`, `${userPrefix}${idx + 2}`];
 }
 
 export function getUserStoragesForTest() {
     const names = getUsernamesForTest();
-    return [`${names[0]}StorageState.json`, `${names[1]}StorageState.json`];
+    return [`${names[0]}StorageState.json`, `${names[1]}StorageState.json`, `${names[2]}StorageState.json`];
+}
+
+export function getUserIDsForTest() {
+    const storageStates = getUserStoragesForTest();
+    return storageStates.map((ss) => {
+        const data = JSON.parse(fs.readFileSync(ss, 'utf-8'));
+        const idx = data.cookies.findIndex((c: { name: string; }) => c.name === 'MMUSERID');
+        return data.cookies[idx].value;
+    });
```

Comment:
That's quite elaborate. We may reach a point where we need to either store these information in advance or use an API client to fetch them but this works too for now :)

Reply:
This one of the instances where I tried a few ways to store them ahead of time and I couldn't make it work so I gave up and did this. I figured it's basically the same as starting a new page or making a request with the userStorage context, you know?

Reply:
I suppose that's the convoluted way Playwright stores state anyway so it's fine :)